File: example-provider/src/main/java/provider/InformationController.java
package provider;

import java.util.HashMap;

import java.util.Map;
import org.springframework.web.bind.annotation.*;

@RestController
public class InformationController {

    @RequestMapping("/information")
    public Information information(@RequestParam(value="name", defaultValue="linkedin") String name) {
        return getSpongeBobData();
    }

    private Information getSpongeBobData()
    {
        Information information = new Information();
        Map<String, String> contact = new HashMap<>();
        contact.put("email", "spongebob@squarepants.com");
        contact.put("phone", "9090950");
        information.setSkills("Complex Quantum Mechanics; Cooking; Philanthropy");
        information.setContact(contact);
        information.setName("SpongeBob SquarePants");
        information.setCredentials("PhD");
        information.setAge("24");
        information.setRelatives("Harold SquarePants; Margaret SquarePants");
        return information;
    }
}

File: provider/src/main/kotlin/com/thatsabug/provider/InformationController.kt
package com.thatsabug.provider


import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class InformationController {
    @RequestMapping("/information")
    fun information(@RequestParam(value = "name", defaultValue = "linkedin") name: String?): Information? {
        return getSpongeBobData()
    }

    private fun getSpongeBobData(): Information {
        return Information(
                "SpongeBob SquarePants",
                "PhD2",
                "Complex Quantum Mechanics; Cooking; Philanthropy",
                null,
                null,
                "24",
                "Harold SquarePants; Margaret SquarePants"
        )
    }
}